### PR TITLE
[MLIR] reverse int8 type's printing logic

### PR DESCRIPTION
**Pull request by yaochengji**
Specializing for 8-bit integers to ensure values are printed as integers in a generic way will cause a bug, see https://github.com/llvm/llvm-project/issues/69310

### CODE REVIEW

**Comment by sogartar** [comment]
LGTM

File: mlir/include/mlir/Dialect/Mesh/IR/MeshBase.td [hunk 1]
```diff
@@ -58,8 +58,8 @@ def MeshSharding : AttrDef<Mesh_Dialect, "MeshSharding"> {
 
   let parameters = (ins
     AttrParameter<"::mlir::SymbolRefAttr", "cluster placed">:$cluster,
-    ArrayRefParameter<"::mlir::DenseI8ArrayAttr">:$split_axes,
-    OptionalArrayRefParameter<"int8_t">:$partial_axes,
+    ArrayRefParameter<"::mlir::DenseI64ArrayAttr">:$split_axes,
```
Comment (joker-eph) [inline mlir/include/mlir/Dialect/Mesh/IR/MeshBase.td:61]: What about DenseI32ArrayAttr?

**Review by joker-eph — APPROVED** [review]

File: mlir/include/mlir/Dialect/Mesh/IR/MeshBase.td (hunk 1 above)
Comment (yaochengji) [reply mlir/include/mlir/Dialect/Mesh/IR/MeshBase.td]: Modified, both `split_axes` and `partial_axes` to int32

**Comment by rorth** [comment]
FWIW, I've tested this patch on both `amd64-pc-solaris2.11` and `sparcv9-sun-solaris2.11` and MLIR/Flang test results are back to normal.  Thanks for the quick fix.

**Comment by joker-eph** [comment]
@yaochengji you most recent commit is authored from a `@hotmial.com` email, there is a typo there.
(Merging with the right email now).

**Comment by yaochengji** [comment]
> @yaochengji you most recent commit is authored from a `@hotmial.com` email, there is a typo there. (Merging with the right email now).

Oh, Thanks! It is now corrected on my dev machine.

**Pull request merged**